syntax = "proto3";

service Simulation {
    rpc Start(Empty) returns (Empty) {}
    rpc Stop(Empty) returns (Empty) {}
    rpc CreateBody(CreateBodyRequest) returns (Empty) {}
    rpc UpdateBody(DeleteBodyRequest) returns (Empty) {}
    rpc DeleteBody(UpdateBodyRequest) returns (Empty) {}
}

message Position {
    float x = 1;
    float y = 2;
}

message Velocity {
    float x = 1;
    float y = 2;
}

message CreateBodyRequest {
    int32 id = 1;
    float mass = 2;
    Position position = 3;
    Velocity velocity = 4;
}

message DeleteBodyRequest {
    int32 id = 1;
}

message UpdateBodyRequest {
    int32 id = 1;
    optional float mass = 2;
    optional Position position = 3;
    optional Velocity velocity = 4;
}

message Empty {}